### PR TITLE
🦔 Alternative Switch to PNPM

### PR DESCRIPTION
**Pull request by rzadp**
Alternative to https://github.com/TrueFiEng/useDApp/pull/560 using `pnpm import`

### CODE REVIEW

File: packages/core/src/providers/network/provider.tsx [hunk 1]
```diff
@@ -51,7 +52,10 @@ export function NetworkProvider({ children }: NetworkProviderProps) {
 
   const activate = useCallback(
     async (provider: JsonRpcProvider | (EventEmitter & ExternalProvider)) => {
-      const wrappedProvider = provider instanceof JsonRpcProvider ? provider : new Web3Provider(provider)
+      const wrappedProvider =
+        provider instanceof JsonRpcProvider || provider instanceof providers.Web3Provider
```
Comment (dmaretskyi) [inline packages/core/src/providers/network/provider.tsx:56]: Can it an already existing import from `@ethersproject/providers`? The `ethers` one should be just a re-export

Comment (rzadp) [reply packages/core/src/providers/network/provider.tsx:56]: It doesn't work when I try that.

Comment (dmaretskyi) [reply packages/core/src/providers/network/provider.tsx:56]: why?

Comment (rzadp) [reply packages/core/src/providers/network/provider.tsx:56]: It's not that instance apparently, it gets wrapped by `new Web3Provider(provider)` and then it crashes with weird errors without stacktracke:

```
Error: Method [object Object] not supported.
```

Comment (dmaretskyi) [reply packages/core/src/providers/network/provider.tsx:56]: looks like there are duplicate dependencies installed 
ethers@5.5.0 vs ethers@5.5.3

**Review by dmaretskyi — APPROVED** [review]

**Pull request merged**